vraptor-core/src/main/java/br/com/caelum/vraptor/observer/ExecuteMethod.java

```diff
@@ -79,6 +85,11 @@ public Void call() throws Exception {
 
 				log.debug("Invoking {}", reflectionMethod);
 				Object instance = event.getControllerInstance();
+				
+				Class<?> declaringClass = reflectionMethod.getDeclaringClass();
+				if (!declaringClass.isAssignableFrom(instance.getClass())) {
+					instance = container.instanceFor(declaringClass);
```

Comment:
I'm wandering if this issue is related with that
dispatcherTypes={DispatcherType.FORWARD,
DispatcherType.REQUEST} filter configuration. It looks like the forwarded
request is not getting into vraptor's filter...

Is this application configuring VRaptor with a web.xml?

On Tue, May 19, 2015 at 1:03 PM Rodrigo Turini <notifications@github.com>
wrote:

> In
> vraptor-core/src/main/java/br/com/caelum/vraptor/observer/ExecuteMethod.java
> <https://github.com/caelum/vraptor4/pull/978#discussion_r30615395>:
>
> > @@ -79,6 +85,11 @@ public Void call() throws Exception {
> >
> >  				log.debug("Invoking {}", reflectionMethod);
> >  				Object instance = event.getControllerInstance();
> > +				
> > +				Class<?> declaringClass = reflectionMethod.getDeclaringClass();
> > +				if (!declaringClass.isAssignableFrom(instance.getClass())) {
> > +					instance = container.instanceFor(declaringClass);
>
> yes, totally. It's already causing another bugs (recently reported here).
>
> A forward to String does a real container forward that's why we get the
> bug?
>
> yes, it does! but I'm sure why it's a problem
>
> I'll take a look on it asap, but any help would be very welcome :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caelum/vraptor4/pull/978/files#r30615395>.
>